ChatSecure/Classes/Utilities/OTRGroupAvatarGenerator.swift

```diff
@@ -0,0 +1,98 @@
+        let range = 0.3 * Double(height)
+        let halfrange = Int(range / 2)
+        let a = Int(generator.random() * range) - halfrange
+        let b = Int(generator.random() * range) - halfrange
+        let c = Int(generator.random() * range) - halfrange
+        let d = Int(generator.random() * range) - halfrange
+        
+        let arrayOfColors:[CGColor] = [
+            colorWithHexString(hexColorString: "#ff58e2c2").cgColor,
+            colorWithHexString(hexColorString: "#fff44058").cgColor,
+            colorWithHexString(hexColorString: "#fff7e53b").cgColor
+        ]
+        let colorTop = arrayOfColors[Int(generator.random().multiplied(by: Double(arrayOfColors.count)))]
+        let colorMiddle = arrayOfColors[Int(generator.random().multiplied(by: Double(arrayOfColors.count)))]
+        let colorBottom = arrayOfColors[Int(generator.random().multiplied(by: Double(arrayOfColors.count)))]
+        
+        UIGraphicsBeginImageContext(CGSize(width: width, height: height))
+        if let ctx = UIGraphicsGetCurrentContext() {
+            // Middle part (just fill all)
+            ctx.setFillColor(colorMiddle)
+            ctx.fill(CGRect(x: 0, y: 0, width: width, height: height))
+
+            let aThird = Int(0.33 * Double(height))
+            let twoThirds = Int(0.66 * Double(height))
+            
+            // Top third
+            ctx.setFillColor(colorTop)
+            ctx.move(to: CGPoint(x: 0, y: 0))
+            ctx.addLine(to: CGPoint(x: 0, y: aThird + a))
+            ctx.addLine(to: CGPoint(x: width, y: aThird + c))
+            ctx.addLine(to: CGPoint(x: width, y: 0))
+            ctx.addLine(to: CGPoint(x: 0, y: 0))
+            ctx.drawPath(using: .fill)
+            
+            // Bottom third
+            ctx.setFillColor(colorBottom)
+            ctx.move(to: CGPoint(x: 0, y: twoThirds + b))
+            ctx.addLine(to: CGPoint(x: 0, y: height))
+            ctx.addLine(to: CGPoint(x: width, y: height))
+            ctx.addLine(to: CGPoint(x: width, y: twoThirds + d))
+            ctx.addLine(to: CGPoint(x: 0, y: twoThirds + b))
+            ctx.drawPath(using: .fill)
+        }
+        let newImage = UIGraphicsGetImageFromCurrentImageContext();
+        UIGraphicsEndImageContext();
+        return newImage!
+    }
+    
+    static func colorWithHexString(hexColorString:String) -> UIColor {
+        let scanner = Scanner(string: hexColorString)
+        scanner.scanLocation = 1
+        var rgb: UInt32 = 0
+        guard scanner.scanHexInt32(&rgb) else { return UIColor.white }
+        let a:CGFloat = CGFloat((rgb & 0xFF000000) >> 24) / 255.0
+        let r:CGFloat = CGFloat((rgb & 0x00FF0000) >> 16) / 255.0
+        let g:CGFloat = CGFloat((rgb & 0x0000FF00) >> 8) / 255.0
+        let b:CGFloat = CGFloat(rgb & 0x000000FF) / 255.0
+        return UIColor(red: r, green: g, blue: b, alpha: a)
+    }
+    
+    // A LCG to create pseudo-random numbers (using a seed)
```

Comment:
avatarImage and colorWithHexString is original N-Pex (the latter has a .m version in Zom), the LCG was inspired by code now linked.